server/src-lib/Hasura/GraphQL/Execute.hs

```diff
@@ -391,67 +390,72 @@ execRemoteGQ
   :: ( MonadIO m
      , MonadError QErr m
      , MonadReader ExecutionCtx m
+     , MonadState QueryLog m
      )
-  => RequestId
-  -> UserInfo
+  => UserInfo
   -> [N.Header]
   -> RemoteSchemaInfo
   -> G.OperationType
   -> VQ.SelSet
   -> m (HttpResponse EncJSON)
-execRemoteGQ reqId userInfo reqHdrs rsi opType selSet = do
+execRemoteGQ userInfo reqHdrs rsi opType selSet = do
   execCtx <- ask
-  let logger  = _ecxLogger execCtx
-      manager = _ecxHttpManager execCtx
+  let manager = _ecxHttpManager execCtx
   when (opType == G.OperationTypeSubscription) $
     throw400 NotSupported "subscription to remote server is not supported"
   hdrs <- getHeadersFromConf hdrConf
   gqlReq <- fieldsToRequest opType (toList selSet)
   let body = encJToLBS $ encJFromJValue gqlReq
-  let confHdrs   = map (\(k, v) -> (CI.mk $ CS.cs k, CS.cs v)) hdrs
+  let confHdrs = map (\(k, v) -> (CI.mk $ CS.cs k, CS.cs v)) hdrs
       clientHdrs = bool [] filteredHeaders fwdClientHdrs
       -- filter out duplicate headers
       -- priority: conf headers > resolved userinfo vars > client headers
-      hdrMaps    = [ Map.fromList confHdrs
-                   , Map.fromList userInfoToHdrs
-                   , Map.fromList clientHdrs
-                   ]
-      headers  = Map.toList $ foldr Map.union Map.empty hdrMaps
+      hdrMaps =
+        [ Map.fromList confHdrs
+        , Map.fromList userInfoToHdrs
+        , Map.fromList clientHdrs
+        ]
+      headers = Map.toList $ foldr Map.union Map.empty hdrMaps
       finalHeaders = addDefaultHeaders headers
   initReqE <- liftIO $ try $ HTTP.parseRequest (show url)
   initReq <- either httpThrow pure initReqE
-  let req = initReq
-           { HTTP.method = "POST"
-           , HTTP.requestHeaders = finalHeaders
-           , HTTP.requestBody = HTTP.RequestBodyLBS body
-           , HTTP.responseTimeout = HTTP.responseTimeoutMicro (timeout * 1000000)
-           }
-
-  -- liftIO $ logGraphqlQuery logger $ QueryLog q Nothing reqId
-  res  <- liftIO $ try $ HTTP.httpLbs req manager
+  let req =
+        initReq
+          { HTTP.method = "POST"
+          , HTTP.requestHeaders = finalHeaders
+          , HTTP.requestBody = HTTP.RequestBodyLBS body
+          , HTTP.responseTimeout = HTTP.responseTimeoutMicro (timeout * 1000000)
+          }
+  -- add to logging
```

Comment:
Maybe a better name than `addToLog`? And "add to logging" isn't a super helpful comment :)